woocommerce/woocommerce-android
# Feature/7254 reduce infinite loading

Closes: #7254 


### Description
This PR aims to provide a better feedback and support when users experience Jetpack tunnel timeout errors when trying to verify their site. Right now this is what they experience: 

https://user-images.githubusercontent.com/2663464/187166517-f7e876f5-42c6-43b8-af88-58a00c54e6fc.mp4


With these changes applied in [FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2512) and the current changes from this PR we will display a new error dialog in SitePicker providing an option to contact support. 
  
### Testing instructions
Testing this is a bit tricky as I haven't found a way to force test sites to make them slow enough to generate timeout errors. I think the easiest way is to: 
- Enable composite build for FluxC by uncommenting `localFluxCPath = "../WordPress-FluxC-Android"` from `local-buidls.gradle` 
-  Checkout branch `feature/7254-reduce-inifinte-loading` in FluxC repo
- Go to `JetpackTunnelGsonRequestBuilder` and set `DEFAULT_JETPACK_TUNNEL_TIMEOUT_MS` to 200ms. This will ensure we get a timeout error whenever trying to verify site 
- Check the following dialog is shown:

https://user-images.githubusercontent.com/2663464/187158794-6bec3a97-e9f7-4921-94ef-0d7e86f3b04e.mp4


- Click on Contact Support and verify Help screen is open with the dialog that starts the create ticket flow. 
- Check the following events are tracked: 
`site_picker_jetpack_timeout_error_shown` 
and 
`site_picker_jetpack_timeout_contact_support_clicked`
